content/programming/joblib_low_overhead_persistence.md

```diff
@@ -8,61 +8,102 @@ Summary: New persistence in joblib enables low-overhead storage of big data cont
 
 ## Joblib persistence improvements
 
-### Introduction
+Joblib is a powerful python package when working on so called __Big Data__, e.g
+data that can consume up to the available RAM (several GB nowadays), generally
+numpy arrays or arbitrary containers (list, dict) of numpy arrays. In order to
+speedup Big Data processing and management, joblib uses a transparent disk
+persistence which is based on functions `dump` and `load`. The underlying
+mecanism is based on an advanded usage of
+[pickle](https://docs.python.org/3/library/pickle.html).
 
-Started to work recently on joblib trying to improve the persistence of numpy
-arrays. 2 problems:
-* memory copies
-* one file per array
+Some [recent work](https://github.com/joblib/joblib/pull/260) on joblib has been
+achieved around memory consumption and compression when dealing with data
+persistence.
 
-Example :
+Up to version 0.9.4, it was known that the process of dumping/loading
+persisted data with compression was a memory gap, mainly because of memory
+copies of numpy arrays, limitating the maximum size of usable data with joblib.
+
+Another drawback was that each numpy array of an arbitrary container were dumped
+in separate `.npy` files, increasing the complexity of the cache management on
+the file system.
+
+Let's explain those problems with a bit of code:
+
+* Container with several numpy arrays is persisted in multiple files:
 ```python
 >>> import numpy as np
 >>> import joblib
->>> obj = [np.arange(10), np.ones((5,5))]
+>>> joblib.__version__
+'0.9.4'
+>>> obj = [np.arange(1000000), np.random.random((10000, 10000))]
 
-# Note that 3 files are generated:
->>> joblib.dump(obj, '/tmp/test.pkl')
-['/tmp/test.pkl', '/tmp/test_01.npy', '/tmp/test_02.npy']
+# 3 files are generated:
+>>> joblib.dump(obj, '/tmp/test.pkl', compress=True)
```

Comment:
My bad, I forgot to update the output : it generates 2 files in fact (`/tmp/test.pkl` and `/tmp/test.pkl_01.npy.z`). The thing is that in 0.9.4 the `cache_size` (which is by default 10MB) parameter partially hide this problem. The `np.arange` goes in `tmp/test/pkl` using pickle because its size is less than 10MB whereas the `np.random.random` goes in a separate file because its size is greater (~700MB).

I'll find better candidates.